src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/ChainPal.cs

```diff
@@ -120,60 +148,86 @@ private SafeCreateHandle PreparePoliciesArray(bool checkRevocation)
             return policiesArray;
         }
 
-        private SafeCreateHandle PrepareCertsArray(ICertificatePal cert, X509Certificate2Collection extraStore)
+        private SafeCreateHandle PrepareCertsArray(
+            ICertificatePal cert,
+            X509Certificate2Collection extraStore,
+            X509Certificate2Collection customTrustStore,
+            X509ChainTrustMode trustMode)
         {
-            IntPtr[] ptrs = new IntPtr[1 + (extraStore?.Count ?? 0)];
-            SafeHandle[] safeHandles = new SafeHandle[ptrs.Length];
-
+            List<SafeHandle> safeHandles = new List<SafeHandle>();
             AppleCertificatePal applePal = (AppleCertificatePal)cert;
-
-            safeHandles[0] = applePal.CertificateHandle;
+            safeHandles.Add(applePal.CertificateHandle);
```

Comment:
Nit: does this need a local?  It could just be:
```C#
safeHandles.Add((AppleCertificatePal)cert).CertificateHandle);
```
or even:
```C#
var safeHandles = new List<SafeHandle> { (AppleCertificatePal)cert).CertificateHandle };
```